root/static/scripts/edit/URLCleanup.js

```diff
@@ -2731,7 +2731,8 @@ const CLEANUPS: CleanupEntries = {
     match: [new RegExp('^(https?://)?([^/]+\\.)?linkedin\\.com/', 'i')],
     restrict: [LINK_TYPES.socialnetwork],
     clean: function (url) {
-      return url.replace(/^https?:\/\/([^/]+\.)?linkedin\.com/, 'https://$1linkedin.com');
+      url = url.replace(/^https?:\/\/(?:[^/]+\.)?linkedin\.com\/([^?#]+)(?:[?#].*)?$/, 'https://www.linkedin.com/$1');
```

Comment:
Correct but can be simplified as follows since the [quantifier](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions/Quantifiers) `+` is greedy (matching the maximum number of times) by default:

```suggestion
      url = url.replace(/^https?:\/\/(?:[^/]+\.)?linkedin\.com\/([^?#]+).*$/, 'https://www.linkedin.com/$1');
```